lib/service/storage/elasticsearch.ts

```diff
@@ -788,14 +869,24 @@ class ElasticSearch extends Service {
    * @returns {Promise.<{ _id, _version }>}
    */
   async update(
-    index,
-    collection,
-    id,
-    content,
-    { refresh, userId = null, retryOnConflict, injectKuzzleMeta = true } = {},
+    index: string,
+    collection: string,
+    id: string,
+    content: JSONObject,
+    {
+      refresh,
+      userId = null,
+      retryOnConflict,
+      injectKuzzleMeta = true,
+    }: {
+      refresh?: boolean | "wait_for";
+      userId?: string;
+      retryOnConflict?: number;
+      injectKuzzleMeta?: boolean;
+    } = {},
```

Comment:
```suggestion
    {
      refresh,
      userId = null,
      retryOnConflict,
      injectKuzzleMeta = true,
    }: Exclude<KRequestParams, "timeout"> = {},
```

With new `KRequestParams` type we can compose type like it, to DRY